### DISCUSSION TITLE
请问有没有可能编译到Windows Arm64系统上？Is it possible to compile Magpie to win arm

### DISCUSSION THREAD

**Discussion by luojunyuan** [Blinue/Magpie#178] [Q&A]
不过转译也能正常运行，放大到2880*1800分辨率帧数在30帧到60帧之间，也可以接受。不太清楚如果Magpie有原生arm版本会不会对性能有较大的提升。我使用的版本是v0.7.0 Preview1

**Comment by Blinue** [comment]
有人尝试过并可能成功了 #161 

移植到 ARM64 上我认为不太可能。一方面 .NET framework 4.7 不能在 ARM64 上运行，另外 Magpie 依赖的第三方库没有 ARM64 版本，你需要自己去编译。

**Reply by Blinue** [reply]
我觉得转译不会对性能有很大影响，它发生在 CPU 上，帧率低可能是受限于你的 GPU 性能

**Reply by luojunyuan** [reply]
issue里的@sqwwqw5 也是转译运行的。Magpie 有可能移植到.net core吗，我看代码csharp的占比并不多，他只是充当gui的作用吗？当然移植arm想必也需要把那些第三方库编译成arm。。繁重的工作。

**Reply by luojunyuan** [reply]
> 我觉得转译不会对性能有很大影响，它发生在 CPU 上，帧率低可能是受限于你的 GPU 性能

应该是这样！我是m1虚拟机，任务管理器中都没有gpu设备，设备管理器中显示的是wddm（可能属于集显？）。游玩文字游戏这个帧率也很舒适了。Magpie真是拯救了一众Galgame😜。

**Reply by Blinue** [reply]
我也有迁移到 .NET Core（或更新版本）的意向，可能会在未来实现。Magpie 使用 Conan 管理 C++ 依赖，使用的库都没有预编译的 ARM 版本，我编码的时候也不会考虑 ARM 平台。

**Reply by Blinue** [reply]
视觉小说类游戏最好的缩放算法是 ACNet，v0.7 的正式版会重新支持。当然它对 GPU 的要求比 FSR 高的多
